Notame normalization and feature selection

vsuksi opened the pull request:
Initial version, using notame 0.3.2, https://github.com/antonvsdata/notame.

philouail commented on vignettes/notame_normalization_and_feature_selection.qmd:
````diff
@@ -0,0 +1,298 @@
+res <- readRDS("res.RDS")
+```
+
+The SummarizedExperiment container used for the data is very similar to the 
````
we usually put function, variables, classes between backticks "``":   `SummarizedExperiment`  `exprs()`   so it's clearer to the reader what is natural language and what is not

philouail commented on vignettes/notame_normalization_and_feature_selection.qmd:
````diff
@@ -0,0 +1,298 @@
+```
+
+# Normalization
+Features with a low detection rate are flagged, as they are deemed too unreliable not only for statistical analysis, but also for the normalization process which relies heavily on QC samples. Flagged features are automatically excluded in *notame*, but can be included using `all_features = TRUE` in many functions. We set the detection rate threshold for QC samples at 70% as per Broadhurst et al.  ...
````
I can see you are quotting a paper, do you have a `reference.bib` file for this vignette ? if yes can you also add it to the github repo ? otherwise no worries I'll build it later on myself. if you add it maybe name it `reference_notame.bib`

philouail commented on vignettes/notame_normalization_and_feature_selection.qmd:
````diff
@@ -0,0 +1,298 @@
+detected <- flag_detection(res, qc_limit = 0.7, group_limit = 0.8)
+```
+
+`r sum(!is.na(flag(detected)))` features were flagged for detection rate, so now we  have reduced the number of features from `r sum(nrow(detected))` to `r nrow(drop_flagged(detected))`. Brief notes about the flagging was stored in feature metadata.
````
Maybe actually put the small code you have in the text within the chunk ? for R beginners these small computation are nice to know about. As the vignette is not so long i don't think it will be overbearing either :)

philouail commented on vignettes/notame_normalization_and_feature_selection.qmd:
````diff
@@ -0,0 +1,298 @@
+drift_normalized <- correct_drift(drop_flagged(detected))
+```
+
+`r sum(fData(drift_normalized)$DC_note == "Drift_corrected")` features were corrected for drift. `r sum(fData(drift_normalized)$DC_note == "Missing_QCS")` features had missing QC values after flagging for detection rate. These were not corrected because at least four QC samples with values are needed for fitting the cubic spline.
````
Same I would also show this code within a chunk instead of text

philouail reviewed (approved):
This is really good ! thank you so so much for taking the time for this. I have added comments for things that need to be updated. 

A general comment would be that if you have parameters specific to notame functions maybe detail them a bit in the text. otherwise I find it very well described. 

I will update the trunk vignette asap so that the `SummarizedExperiment` is saved and therefore your vignette can work with everything else ! 

Request my feedback (button on the top right) when you have done the little fixes.

philouail commented on vignettes/notame_normalization_and_feature_selection.qmd:
```diff
@@ -0,0 +1,298 @@
+library(notame)
+library(SummarizedExperiment)
+
+registerDoParallel(cores = 14)
```
and therefore no need for the `doParallel` package

jorainer replied:
Actually, I have a question here @vsuksi - I assume you use `doParallel` parallel processing code within the `notame` package? Also thinking on the Bioconductor submission of your package, I think it would be good to change/switch here to `BiocParallel` parallel processing functions (i.e. `bplapply()`, `bpmapply()` etc). The `BiocParallel` is the *standard* package in Bioconductor to register/configure parallel processing. So, something like the example code above from Phili would then be used across all used packages for the parallel processing. What I found particularly helpful with `BiocParallel` is that it allows to use a variety of different configurations, from `SerialParam` to disable parallel processing, to `SnowParam` (for Windows) `MulticoreParam` (for Unix) to also parameters that allow to configure and use HPC queueing systems like slurm. And there is also a param for `doParallel` too :)

So, maybe, on the longer run, it would make sense to switch the parallel processing code within `notame` to `BiocParallel`?

vsuksi replied:
Good to hear and sorry for the delay, I need to practice juggling several things at once! Happily the Bioconductor-version of notame uses BiocParallel instead of doParallel, I will switch to that version and the code chunk above after we're done with some larger changes and submit.  We've been taking some time to make this the first real release, but I'm hoping it is ready in a week or two. 

Do you want all references in the same .bib file or should I include a new one for notame normalization and feature selection?

philouail commented:
Just converting to draft so that we don't merge per error until it's ready :)

vsuksi commented:
Hi, I think I've addressed your suggestions, thanks! I also changed some minor things and clarified in the normalization visualizations what one would typically look for.

philouail commented on vignettes/notame-references.bib:
```diff
@@ -0,0 +1,42 @@
+@article{klavus_notame_2020,
```
This is perfect thanks a lot !

philouail commented:
Indeed this is completely good now ! If the next version of *notame* with `BiocParallel` comes out in a few weeks, let's wait until then before merging :) I am fixing some error that arise after the latest BioC update anyway so there is no rush. 

Thanks a lot again ! Super excited to add it to the website :)

philouail commented on vignettes/notame_normalization_and_feature_selection.qmd:
```diff
@@ -0,0 +1,315 @@
+
+registerDoParallel(cores = 14)
+
+dir <- system.file("extdata", "preprocessed_res", package = "metabonaut")
```
Hi sorry just tiny thing that needs to change: can you write `Metabonaut` with a capital letter instead of `metabonaut`

philouail commented:
FYI, I have now an "official" version 1.0.0 of Metabonaut. So I implemented a "devel" branch for the new implementation. so if possible when you update this, close it and do a PR to the "devel" branch instead :)
